frame/evm/src/lib.rs

```diff
@@ -677,6 +678,29 @@ pub mod pallet {
 
 	#[pallet::storage]
 	pub type Suicided<T: Config> = StorageMap<_, Blake2_128Concat, H160, (), OptionQuery>;
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_: BlockNumberFor<T>) -> Weight {
+			let pov_before = get_proof_size();
```

Comment:
Should the logic be wrapped in a `if let Some(pov_before) = get_proof_size()` ? as `get_proof_size` returns `None` if proof recording is disabled on the host, so next call would also return `None`.